File: late.py
#!/usr/bin/env python3
"""
Late penalty calculator using Python 3
Jim Teresco, The College of Saint Rose, CSC 433, Fall 2014
"""

penalty = 0.0
lateness = 1
while (penalty < 100.0):
    penalty = 1.08**lateness
    print('Hour ' + str(lateness) + ' late = %4.2f%% penalty' % penalty)
    lateness = lateness + 1

File: late.pl
#!/usr/bin/perl
#
# Script to demonstrate late penalties
#
# Jim Teresco
# The College of Saint Rose
# CSC 433, Programming Languages, Fall 2012
#
$base = 1.08;
$lateness = 1;
$penalty = 0.0;
while ($penalty < 100) {
    $penalty = $base**$lateness;
    printf("Hour $lateness late = $penalty%% penalty\n");
    $lateness++;
}
